src/main/java/ladder/domain/LadderResult.java

```diff
@@ -1,43 +1,40 @@
 package ladder.domain;
 
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 public class LadderResult {
     private static final String KEY_ERROR = "참여자를 잘못 입력했습니다.";
     private static final int DEFAULT_INDEX = 0;
-    private final Map<String, String> results;
+    private final Map<Person, Winning> results;
 
     private LadderResult(Map results) {
         this.results = results;
     }
 
-    public static LadderResult of(Person person, Winning winning, Ladder ladder) {
-        Map<String, String> result = new HashMap<>();
+    public static LadderResult of(Persons person, Winnings winning, Ladder ladder) {
+        Map<Person, Winning> result = new HashMap<>();
         int[] lineIndex = {DEFAULT_INDEX};
-        person.readOnlyPerson().stream()
+        person.readOnlyPersons()
                 .forEach(index -> result.put(index, ladderResult(lineIndex[DEFAULT_INDEX]++, winning, ladder)));
         
         return new LadderResult(result);
     }
 
-    private static String ladderResult(int nameIndex, Winning winning, Ladder ladder) {
+    private static Winning ladderResult(int nameIndex, Winnings winning, Ladder ladder) {
         int[] lineIndex = { nameIndex };
-        ladder.readOnlyLines().stream()
+        ladder.readOnlyLines()
                 .forEach(line -> lineIndex[DEFAULT_INDEX] = line.lineMoving(lineIndex[DEFAULT_INDEX]));
 
-        return winning.readOnlyWinning().get(lineIndex[DEFAULT_INDEX]);
+        return winning.findWinning(lineIndex[DEFAULT_INDEX]);
     }
 
-    public String resultOfLadder(String name) {
-        if (!results.containsKey(name)) {
-            throw new IllegalArgumentException(KEY_ERROR);
-        }
-        return results.get(name);
-    }
+    public Winning resultOfLadder(String name) {
+        Person person1 = results.keySet().stream()
+                .filter(person -> person.isName(name))
```

Comment:
아무래도 주셨던 질문이 해당 코드를 작성하시면서 의문이 들었던 것 같아요!
https://github.com/next-step/java-ladder/pull/826#issuecomment-809345846

이번 상황과 같이 기능에 따라서 개발자가 특별한 Key에 대한 타입을 정의해야할 필요가 있습니다.
(특히 웹개발을 하다보면 데이터들을 VO객체에 담거나 Map 객체에 key - value로 세팅하는 경우도 많죠)
현재 name으로 key를 찾고 있는 구조네요 ㅎㅎ 만약 필드가 많아지면 그만큼 불편한 점도 있을 것 같아요 ㅎㅎ

다음과 같이 객체에게 동일한 값을 갖고있는지 물어보는 메서드를 만들지말고
논리적인 동치성을 비교하도록 하면 어떨까요 :) ?
Winning 메서드에서는 잘해주셨네요 👍 

```suggestion
    public Winning resultOfLadder2(Person person) {
        if (results.containsKey(person)) {
            return results.get(person);
        }
        throw new IllegalArgumentException(KEY_ERROR);
    }
```